src/core/server/http/integration_tests/router.test.ts

```diff
@@ -46,6 +46,177 @@ afterEach(async () => {
   await server.stop();
 });
 
+describe('Options', () => {
+  describe('authRequired', () => {
+    describe('optional', () => {
+      it('User has access to a route if auth mechanism not registered', async () => {
```

Comment:
Should we verify the value of `context.auth.isAuthenticated` in these tests?

Reply:
we have separate tests for them, but it doesn't hurt since here we can verify:
- a route was hit
- auth API returns a correct result